### PR TITLE
clang check modernize-use-emplace

### PR DESCRIPTION
**Pull request by davidlange6**
for comment about what clang-tidy does to CMSSW (and likely big and not mergeable - this is what clang-tidy check modernize-use-emplace does on the tip of CMSSW_9_3_X as of a few days ago

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: DataFormats/TauReco/src/PFTau3ProngSummary.cc [hunk 1]
```diff
@@ -10,10 +10,10 @@ PFTau3ProngSummary::PFTau3ProngSummary(reco::PFTauTransverseImpactParameterRef T
     has3ProngSolution_.push_back(false);
     solution_Chi2_.push_back(0);
     thetaGJsig_.push_back(0);
-    tau_p4_.push_back(TLorentzVector(0,0,0,0));
-    daughter_PDGID_.push_back(std::vector<int>());
-    daughter_charge_.push_back(std::vector<int>());
-    daughter_p4_.push_back(std::vector<TLorentzVector>());
+    tau_p4_.emplace_back(0,0,0,0);
```
Comment (slava77) [inline DataFormats/TauReco/src/PFTau3ProngSummary.cc]: it may be still more readable to keep the type in place for cases with non-trivial constructors

**Comment by slava77** [comment]
-1

- OK for cases like ```push_back(x) -> emplace_back(x)```
- OK for in place construction with under 2 arguments e.g. ```push_back(someType(literal)) -> emplace_back(literal)```
- more complex calls appear to complicate readability

**Comment by ianna** [comment]
@davidlange6 - there are 32 Geometry packages which used to use push_back (see https://github.com/cms-sw/cmssw/pull/19975), but this PR touches only 15. Is there a specific selection?

**Comment by davidlange6** [comment]
The tool was applied to all CMSSW packages. However, I don't believe this tool is meant to replace all occurrences of push_back.

**Comment by ianna** [comment]
-1

just to remove from geometry list

Please use the following PRs instead:
https://github.com/cms-sw/cmssw/pull/19976
https://github.com/cms-sw/cmssw/pull/19975

**Pull request closed without merging**